components/identity/org.wso2.carbon.identity.certificateauthority/src/main/java/org/wso2/carbon/identity/certificateauthority/CAAdminService.java

```
@@ -0,0 +1,288 @@
+ */
+@SuppressWarnings("UnusedDeclaration")
+public class CAAdminService extends AbstractAdmin {
+    private static final Log log = LogFactory.getLog(CAAdminService.class);
+
+    /**
+     * DAO for CSR related operations.
+     */
+    private CSRService csrService = CAServiceComponent.getCsrService();
+    private CertificateService certificateService = CAServiceComponent.getCertificateService();
+    private SCEPService scepService = CAServiceComponent.getScepService();
+    private CRLService crlService = CAServiceComponent.getCrlService();
+    private CAConfigurationService configurationService = CAServiceComponent.getCaConfigurationService();
+
+    /**
+     * Get the list of CSR assigned to the current tenant.
+     *
+     * @return list of CSR assigned to the current tenant
+     */
+    public CSR[] listCSRs() throws CAException {
+        try {
+            String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            List<CSR> csrList = csrService.listCSRs(tenantDomain);
+            return csrList.toArray(new CSR[csrList.size()]);
+        } catch (CAException e) {
+            log.error("Error listing the CSRs", e);
+            throw new CAException("Error listing the CSRs");
+        }
+    }
+
+    /**
+     * Gets the CSRs for the tenant CA having the given status.
+     *
+     * @param status The status filter
+     * @return CSRs of the tenant CA which has the given status
+     * @throws CAException
+     */
+    public CSR[] listCSRsByStatus(String status) throws CAException {
+        try {
+            String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            List<CSR> csrList = csrService.listCSRsByStatus(tenantDomain, status);
+            return csrList.toArray(new CSR[csrList.size()]);
+        } catch (CAException e) {
+            log.error("Error when listing CSRs with status " + status, e);
+            throw new CAException("Error when listing CSRs");
```

The original exception is swallowed here. Please fix.